rosetta-source/src/main/rosetta/product-template-type.rosetta

```diff
@@ -23,16 +23,17 @@ import cdm.product.collateral.*
 
 import cdm.mapping.config.*
 
-type TransferableProduct:  <"A TransferableProduct is a type of Product which can be used in a SettlementCommitment for a basic cash settled trade of either an Asset with or without the addition of specific EconomicTerms.">
+type ProductBase: <"Serves as an abstract class to specify a product using a productIdentifier.">
+    productTaxonomy ProductTaxonomy (0..*) <"Specifies the product taxonomy, which is composed of a taxonomy value and a taxonomy source.">
+    economicTerms EconomicTerms (1..1)  <"The price forming features, including payouts and provisions.">
+
+type TransferableProduct extends ProductBase:  <"A TransferableProduct is a type of Product which can be used in a SettlementCommitment for a basic cash settled trade of either an Asset with or without the addition of specific EconomicTerms.">
```

Comment:
Consider instead an inheritance swap:
```
type TransferableProduct extends Asset
```
with `economicTerms` as the "enrichment" attribute.

Advantage: any `Underlier` would have all the common attributes of `Asset`. And in future, `Product` as a choice would still contain `EconomicTerms` in both cases.

`TransferableProduct` probably doesn't need `productTaxonomy`, since it already has `taxonomy` from `Asset`.

Reply:
done